### DISCUSSION TITLE
div directly containing one or more images

### DISCUSSION THREAD

**Discussion by PlethoraChutney** [quarto-dev/quarto-cli#3136] [Q&A] [answered]
Hello, I'm looking for a way to create a div with which directly contains images. One smaller problem is that Quarto wraps the images in `<p>` tags, even when I don't include a caption. Obviously the caption creates a wrapping <figure> structure, which would be fine if I could avoid it by not including a caption.

More importantly, if the wrapping `<div>` has the `.r-stretch` class, the `<div>` disappears and `.r-stretch` is applied to the `<img>` **only when there is a single `<img>`**:

Example:
```
---
title: "DivExample"
author: "Rich Posert"
format: revealjs
---

## Title

::: {.image-container .another-class}

![](https://quarto.org/quarto.png)

:::

## Another title

::: {.r-stretch .image-container .another-class}

![](https://quarto.org/quarto.png)

:::

## A third title

::: {.r-stretch .image-container .another-class}

![](https://quarto.org/quarto.png)
![](https://quarto.org/quarto.png)

:::
```

Expected/Desired:

```
section
    h2
    div.image-container.another-class
        img
section
    h2
    div.r-stretch.image-container.another-class
        img
section
    h2
    div.r-stretch.image-container.another-class
        img
        img
```

Result:

```
section
    h2
    div.image-container.another-class
        p
            img
section
    h2
    img.r-stretch
section
    h2
    div.r-stretch.image-container.another-class
        p
            img
            img
```

**Comment by cscheid** [comment, marked as answer]
## A workaround

If you're looking for a specific DOM structure like this, then a workaround might be to use raw html. Instead of

```
## Title

::: {.image-container .another-class}

![](https://quarto.org/quarto.png)

:::
```

You can write

```
## Title

<div class="image-container another-class"><img src="https://quarto.org/quarto.png"/></div>
```

This does mean that Pandoc filters will not recognize that div, but that might be acceptable in your case.

## Your actual question

Now to your actual question: I think what's happening is that quarto is getting overeager in detecting your image as a _figure_, and then inserting all the scaffolding necessary for captions, labels, etc (eg. the paragraph you're seeing). It would be good if that treatment were more robust, but we're in the process of overhauling that entire system for 1.3, so I don't want to try and create a partial fix right now.

**Reply by PlethoraChutney** [reply]
Ah, I had tried `<div> ![](url) </div>` but for some reason hadn't thought to do everything with HTML. Makes sense! Thank you! Loving Quarto so far!
